src/mds/journal.cc

```diff
@@ -237,6 +237,13 @@ void LogSegment::try_to_expire(MDS *mds, MDSGatherBuilder &gather_bld, int op_pr
     mds->sessionmap.save(gather_bld.new_sub(), sessionmapv);
   }
 
+  // updates to sessions for completed_requests
+  for (std::set<entity_name_t>::iterator i = touched_sessions.begin();
+       i != touched_sessions.end(); ++i) {
+    mds->sessionmap.save_if_dirty(*i, &gather_bld);
+  }
+  touched_sessions.clear();
+
```

Comment:
one op for each dirty session, this seems costly. could you make SessionMap::save() also save touched_session.